sklearn/linear_model/stochastic_gradient.py

```diff
@@ -700,17 +765,42 @@ class SGDClassifier(BaseSGDClassifier):
         - 'constant': eta = eta0
         - 'optimal': eta = 1.0 / (alpha * (t + t0)) [default]
         - 'invscaling': eta = eta0 / pow(t, power_t)
+        - 'adaptive': eta = eta0, as long as the training keeps decreasing.
+        Each time n_iter_no_change consecutive epochs fail to decrease the
+        training loss by tol, or fail to increase validation score by tol if
+        'early_stopping' is on, the current learning rate is divided by 5.
 
         where t0 is chosen by a heuristic proposed by Leon Bottou.
 
     eta0 : double
-        The initial learning rate for the 'constant' or 'invscaling'
-        schedules. The default value is 0.0 as eta0 is not used by the
-        default schedule 'optimal'.
+        The initial learning rate for the 'constant', 'invscaling' or
+        'adaptive' schedules. The default value is 0.0 as eta0 is not used by
+        the default schedule 'optimal'.
 
     power_t : double
         The exponent for inverse scaling learning rate [default 0.5].
 
+    early_stopping : bool, default False
+        Whether to use early stopping to terminate training when validation
+        score is not improving. If set to true, it will automatically set aside
+        a fraction of training data as validation and terminate training when
+        validation score is not improving by at least tol for
+        n_iter_no_change consecutive epochs.
+
+        .. versionadded:: 0.20
+
+    n_iter_no_change : int, default 2
+        Number of iterations with no improvement to wait before early stopping.
+
+        .. versionadded:: 0.20
+
+    validation_fraction : float, optional, default 0.1
+        The proportion of training data to set aside as validation set for
+        early stopping. Must be between 0 and 1.
+        Only used if early_stopping is True
```

Comment:
Note that this proportion will be taken out of each batch to `partial_fit` (which is a bit counter-intuitive)